pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -304,8 +304,8 @@ stdenv.mkDerivation (finalAttrs: {
         if dl.pkg == null then ''
           # remove the dependency on the library by replacing it with an invalid path
           for file in $(grep -lr '"${dl.name}"' src); do
-            echo "patching dlopen(\"${dl.name}\", …) in $file to an invalid store path ("/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-not-implemented/${dl.name}")…"
-            substituteInPlace "$file" --replace '"${dl.name}"' '"/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-not-implemented/${dl.name}"'
+            echo "patching dlopen(\"${dl.name}\", …) in $file to an invalid store path ("${builtins.storeDir}/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-not-implemented/${dl.name}")…"
```

Comment:
Making a comment storeDir-dependent in `systemd` doesn't feel right...

Reply:
> Making a comment storeDir-dependent in `systemd` doesn't feel right: this mainly means that the user has to rebuild everything...

the cache already depends on storeDir so everything has to be rebuilt anyways